tests/functional/test_public_apis.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import mock
+
+from botocore.session import Session
+from botocore.exceptions import NoCredentialsError
+
+
+class EarlyExit(BaseException):
+    pass
+
+def _test_public_apis_will_not_be_signed(func, kwargs):
+    with mock.patch('botocore.endpoint.Session.send') as _send:
+        _send.side_effect = EarlyExit("we don't care about response here")
+        try:
+            func(**kwargs)
+        except EarlyExit:
+            pass
+        except NoCredentialsError:
+            assert False, "NoCredentialsError should not be triggered"
+        request = _send.call_args[0][0]
+        SigV2_disabled = 'SignatureVersion=2' not in request.url
+        SigV3_disabled = 'X-Amzn-Authorization' not in request.headers
+        SigV4_disabled = 'Authorization' not in request.headers
+        assert SigV2_disabled and SigV3_disabled and SigV4_disabled
```

Comment:
Will be changed into:

        SigV2_disabled = 'SignatureVersion=2' not in request.url
        assert SigV2_disabled, "SigV2 is incorrectly enabled"
        SigV3_disabled = 'X-Amzn-Authorization' not in request.headers
        assert SigV3_disabled, "SigV3 is incorrectly enabled"
        SigV4_disabled = 'Authorization' not in request.headers
        assert SigV4_disabled, "SigV4 is incorrectly enabled"